docs/experiments/focus-metavariable.md

```diff
@@ -0,0 +1,27 @@
+---
+slug: focus-metavariable
+append_help_link: true
+description: "focus-metavariable allows Semgrep users to narrow down the code matched by a rule, to focus on what it really matters."
+---
+
+# `focus-metavariable`
+
+The experimental `focus-metavariable` operator puts the focus, or "zooms in", on the code region matched by a metavariable. For example, if we wanted to find all functions arguments annotated with the type `bad` we could write this pattern:
+
+```yaml
+pattern: |
+  def $FUNC(..., $ARG : bad, ...):
+    ...
+```
+
+This works but it matches the entire function definition. Sometimes that is not desirable, for instance, if we are using [Semgrep App](https://semgrep.dev/login) and we have triaged a finding generated by this pattern, the same finding may show up again as new if we make any change to the definition of the function!
+
+Using `focus-metavariable` we can tell Semgrep that we are only interested in the code matched by a particular metavariable, in our case, `$ARG`:
+
+<iframe src="https://semgrep.dev/embed/editor?snippet=4kk8" border="0" frameBorder="0" width="100%" height="435"></iframe>
+
+Note that `focus-metavariable: $ARG` is not the same as `pattern: $ARG`! Using `pattern: $ARG` will find all the uses of the parameter `x` which is not what we want! (Note that `pattern: $ARG` does not match the formal parameter declaration, because in this context `$ARG` only matches expressions.)
+
+<iframe src="https://semgrep.dev/embed/editor?snippet=PPPe" border="0" frameBorder="0" width="100%" height="435"></iframe>
+
+In short, `focus-metavariable: $X` is not a pattern in itself, it does not perform any matching, it just focuses the matching on the code already bound to `$X` by other patterns. Whereas `pattern: $X` will match `$X` against your code (and in this context, `$X` only matches expressions)!
```

Comment:
```suggestion
In short, `focus-metavariable: $X` is not a pattern in itself, it does not perform any matching, it only focuses the matching on the code already bound to `$X` by other patterns. Whereas `pattern: $X` matches `$X` against your code (and in this context, `$X` only matches expressions)!
```

Reply:
Just a note for the use of future tense: https://www.notion.so/r2cdev/r2c-technical-communication-style-guide-4d1104e78927460aa8c7624aedd6561f#ca0bfefb2cfb42339cacc8eeb8a3554f